switchio/handlers.py

```diff
@@ -173,78 +159,63 @@ def _handle_bj(self, e):
         err = '-ERR'
         job_uuid = e.get('Job-UUID')
         body = e.get('Body')
+
         # always report errors even for jobs which we aren't tracking
         if err in body:
             resp = body.strip(err).strip()
             error = True
             self.log.debug("job '{}' failed with:\n{}".format(
                            job_uuid, str(body)))
+        elif ok in body:
+            resp = body.strip(ok + '\n')
 
-        if job_uuid in self.bg_jobs:
-            job = self.bg_jobs.get(job_uuid, None)
+        job = self.bg_jobs.get(job_uuid, None)
+        if not job:
+            job = Job(event=e)
         else:
-            # might be in the middle of inserting a job
-            self._lookup_blocker.wait()
-            job = self.bg_jobs.get(job_uuid, None)
+            job.events.update(e)
 
-        # if this job is registered, process it
-        if job:
-            job.update(e)
-            consumed = True
+        # attempt to lookup an associated session
+        sess = self.sessions.get(job.sess_uuid or resp, None)
+
+        if error:
             # if the job returned an error, report it and remove the job
-            if error:
-                # if this job corresponds to a tracked session then
-                # remove it as well
+            self.log.error(
+                "Job '{}' corresponding to session '{}'"
```

Comment:
Any reason to use .format() when log.error() can do formatting itself? (e.g `log.error('msg %s', arg1)`)

Reply:
Just saw a talk on this at pycon by one of the core devs actually - the `%s` usage is greatly frowned upon due to poor readability.

If I wanted to get fancy I could use the new [f-strings](https://www.python.org/dev/peps/pep-0498/) but I want to keep `py35` support for now / `pypy`.